### DISCUSSION TITLE
Docker isn't updated with patch updates

### DISCUSSION THREAD

**Discussion by ericofusco** [renovatebot/renovate#18742] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.12.0

### If you're self-hosting Renovate, select which platform you are using.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

GitLab (.com and CE/EE)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

Maybe I didn't configure correctly but I'm trying to configure renovate so container images for python are only updated when patch versions are available.

I have the following Dockerfile:

```
FROM python:3.9.1-alpine as builder
```

I tried to skip minor versions with the following packageRules config.

```
    packageRules: [
        {
            matchDatasources: ["docker"],
            matchPackageNames: ["python"],
            matchUpdateTypes: ["minor"],
            enabled: false
        }
    ]
```

As of Today, I'd expect a merge request updating to `3.9.15-alpine`, but it doesn't raise any merge requests, it only raises a merge request when I remove the above packageRules config but it tries to update to python `3.11.0`.

### Relevant debug logs

<details><summary>Logs</summary>

```
    "config": {
         "dockerfile": [
           {
             "packageFile": "docker/Dockerfile.python",
             "deps": [
               {
                 "depName": "python",
                 "currentValue": "3.9.1-alpine",
                 "replaceString": "python:3.9.1-alpine",
                 "autoReplaceStringTemplate": "{{depName}}{{#if newValue}}:{{newValue}}{{/if}}{{#if newDigest}}@{{newDigest}}{{/if}}",
                 "datasource": "docker",
                 "depType": "final",
                 "depIndex": 0,
                 "updates": [
                   {
                     "bucket": "non-major",
                     "newVersion": "3.11.0",
                     "newValue": "3.11.0-alpine",
                     "newMajor": 3,
                     "newMinor": 11,
                     "updateType": "minor",
                     "branchName": "renovate/python-3.x"
                   }
                 ],
                 "warnings": [],
                 "versioning": "docker",
                 "currentVersion": "3.9.1",
                 "isSingleVersion": true,
                 "fixedVersion": "3.9.1-alpine"
               }
             ]
           }
         ]
       }
```
</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
checkout separate minor patch in the docs

**Comment by ericofusco** [comment]
Thanks a lot @viceice , that did the trick.

Posting the relevant config in case it helps folks with a similar issue:

```
    packageRules: [
        {
            matchDatasources: ["docker"],
            matchPackageNames: ["python"],
            separateMinorPatch: true,
        },
        {
            matchDatasources: ["docker"],
            matchPackageNames: ["python"],
            matchUpdateTypes: ["minor"],
            enabled: false
        }
    ]
```
